Makefile

```diff
@@ -0,0 +1,32 @@
+.DEFAULT-GOAL := help
+
+# Variables
+PROJECT_NAME?=$$(git remote -v | head -n1 | awk '{print $$2}' | sed 's/.*\///' | sed 's/\.git//')
+CWD=$(shell pwd)
+VERSION_NUMBER?=$$(docker run -v $(CWD):/repo codacy/ci-git-version:latest)## lazy. run with VERSION_NUMBER=x to prevent slow runs
```

Comment:
What does `lazy. run with VERSION_NUMBER=x to prevent slow runs` mean?

Reply:
VERSION_NUMBER is a lazy variable (the ?= makes it lazy). The command is only evaluated if the VERSION_NUMBER is not defined.
Example:
```
$ make get-next-version-number VERSION_NUMBER=my_version
my_version
```